File: algorithms/python3/119_pascals_triangle_II.py
from typing import List
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        ans=[1]
        for i in range(rowIndex):
            cur_res=ans.copy()
            ans=[1]
            for j in range(len(cur_res)-1):
                ans.append(cur_res[j]+cur_res[j+1])
            ans.append(1)
        return ans

a=Solution()
print(a.getRow(2))

File: algorithms/cpp/119_pascals_triangle_II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int> > res={vector<int> (rowIndex+1), vector<int> (rowIndex+1)};
        for (int i=0;i<=rowIndex;i++) {
            res[i&1][0]=res[i&1][rowIndex]=1;
            for (int j=1;j<rowIndex;j++) {
                res[i&1][j]=res[(i+1) &1][j-1]+res[(i+1)&1][j];
            }
    }
    return res[rowIndex&1];
    }
};
